### DISCUSSION TITLE
Addhole ignoriert AAAA/IPv6

### DISCUSSION THREAD

**Discussion by electronicm** [Freetz-NG/freetz-ng#623] [Allgemein] [answered]
Kann es sein, dass dnsmasq nur A-Querys mit der Host-Liste von Addhole behandelt?
In meinen Logs tauchen nach einem "geblockten" A-Query idR ein AAAA-Query zur selben Domain auf, welcher dann weiter gereicht wird. Somit wird jegliche Werbung, die per IPv4 nicht erreichbar ist, per IPv6 geladen.

Hat das auch schon jmd beobachtet bzw. hat nen Tipp, was bei mir falsch läuft?

**Comment by fda77** [comment]
Wenn ein host keine ipv6 hat kann der dns-server auch keine zurückgeben. Das ist das ganz normale "addn-hosts" von dnsmasq
Probier doch mit "dig" aus ob es in Problem.
Ausserdem kannst du auch ne ip selbst setzen, das muss keine v4 sein

**Reply by electronicm** [reply]
Soweit bin ich mit der Materie vertraut.

Bei mir sind im gesamten Netz IPv4 & IPv6 aktiv, konfiguriert und funktional.
Meine Beobachtungen entspricht folgendem Beispiel/ablauf:
- Client fragt example.com an
- dnsmasq wendet Filter-Regel für A-Record an und antwortet mit 0.0.0.0
- dnsmasq ignoriert Filter-Regel für AAAA-Record, macht nen DNS-Forward und antwortet dann mit einer tatsächlichen IPv6-Adresse

Somit erhält der Client schlussendlich doch Zugriff auf den eigentlich zu blockierenden Content.

**Comment by fda77** [comment]
Das würde mich sehr wundern und wäre ein Bug in dnsmasq. 
es wird einem hostname eine ip gegeben die zurückgegeben werden soll. Das hat nichts mit ipvX zu tun

**Comment by mfamphlett** [comment]
I am seeing the same problem.

If I set "Target IP for blocked hosts:" to "::" then IPv6 is blocked but v4 is not.

Is there any good way to block both v4 and v6 addresses?

I can add 2 entries for each domain, but that seems like a bad solution.

**Comment by fda77** [comment]
Das würde mich sehr wundern und wäre ein Bug in dnsmasq.
es wird einem hostname eine ip gegeben die zurückgegeben werden soll. Das hat nichts mit ipvX zu tun

**Reply by mfamphlett** [reply]
They are 2 different queries A vs AAAA

If there is no configuration for AAAA, then it will be resolved upstream.

As far as I can see, the addhole plugin just adds a bunch of host entries:

```
0.0.0.0 block.this.domain
```

dnsmasq would return 0.0.0.0 for an A query, but for an AAAA it would look upstream as it is not defined.

You would need the following host to also blcok the AAAA in this case:

```
0.0.0.0 block.this.domain
:: block.this.domain
```

**Comment by fda77** [comment]
There is a problem if a host in the blocklist has upstream an A and AAAA record, then dnsmasq answers the A with 0.0.0.0 and the AAAA are the 1 or multiple upstream ips

**Comment by fda77** [comment, marked as answer]
@mfamphlett Please test 320fca777

The count of hosts should be the double now...

What do you think, should be the count of hosts be divided by /2 ?
https://github.com/Freetz-NG/freetz-ng/blob/master/make/pkgs/addhole/files/root/etc/init.d/rc.addhole#L67

**Comment by fda77** [comment]
12767fbc1
